app/views/data_element_values/_form.html.erb

```diff
@@ -1,29 +1,89 @@
 <%= form_for @data_element_value, :html => { :class => 'form-horizontal form-model' } do |f| %>
   <div class="form-group">
-    <%= f.label :environment, :class => 'control-label col-md-2' %>
-    <div class="col-md-10">
+    <%= f.label :environment, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.collection_select :environment_id, @environments, :id, :name, {:class => 'text-field form-control'} %>
     </div>
   </div>
   <div class="form-group">
-    <%= f.label :variable, :class => 'control-label col-md-2' %>
-    <div class="col-md-10">
+    <%= f.label :variable, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.collection_select :data_element_id, @variables, :id, :key, {:class => 'text-field form-control'} %>
     </div>
   </div>
-  <div class="form-group">
-    <%= f.label :value, :class => 'control-label col-md-2' %>
-    <div class="col-md-10">
+  <div class="form-group" id="variable-text-entry">
+    <%= f.label :value, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.text_field :value, :class => 'text-field form-control' %>
     </div>
   </div>
+  <div class="form-group variable-control-container" id="variable-length-entry">
+    <%= f.label :random_value_length, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
+      <%= f.number_field :random_value_length,
+          :class => 'text-field form-control' %>
```

Comment:
It's possible to exceed the max length of 256. Submitting a value this large returns an error screen but the alignment is out.

Preference would be to cap input here at the min/max values allowed.